synthesizer/src/vm/verify.rs

```diff
@@ -149,17 +149,61 @@ impl<N: Network, C: ConsensusStorage<N>> VM<N, C> {
             Transaction::Deploy(id, deployment_id, owner, deployment, _) => {
                 // Verify the signature corresponds to the transaction ID.
                 ensure!(owner.verify(*deployment_id), "Invalid owner signature for deployment transaction '{id}'");
-                // Ensure the edition is correct.
-                if deployment.edition() != N::EDITION {
-                    bail!("Invalid deployment transaction '{id}' - expected edition {}", N::EDITION)
-                }
-                // Ensure the program ID does not already exist in the store.
-                if self.transaction_store().contains_program_id(deployment.program_id())? {
-                    bail!("Program ID '{}' is already deployed", deployment.program_id())
+                // Verify the program checksum, if it exists.
+                // TODO (@d0cd): Add requirement for the program checksum after migration height. Use similar logic to how the execution fee is checked.
```

Comment:
Migration height check for checksum might be sufficient since they are coupled, but we may want to check the absence of a constructor as well if the migration height is not reached.

Reply:
Good point, I believe it's needed

Reply:
I have an attempt migration PR here. It needs work as setting V5 height invalidates many of tests that use constructors.